tree.go

```diff
@@ -0,0 +1,41 @@
+package fastmerkle
+
+// Node represents a single node in the Merkle
+type Node struct {
+	Hash []byte // The hash of the children nodes
+
+	Left   *Node // Reference to the left child
+	Right  *Node // Reference to the right child
+	Parent *Node // Reference to the parent node
```

Comment:
Actually are these fields used?

Reply:
Not yet, but they will be heavily used when we add proof verification + generation